### PR TITLE
Loadd cursor

### PR DESCRIPTION
**Pull request by MilesMcBain**
# Summary

An RStudio addin to load the target identified by the symbol at the current editor cursor position to the global environment. 

# Related GitHub issues and pull requests

- Ref: #763

# Checklist

- [ *] I have read `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CODE_OF_CONDUCT.md), and I agree to follow its rules.
- [*] I have listed any substantial changes in the [development news](https://github.com/ropensci/drake/blob/master/NEWS.md).
- [ *] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/ropensci/drake/tree/master/tests/testthat) to confirm that any new features or functionality work correctly.
- [ *] I have tested this pull request locally with `devtools::check()`
- [ *] This pull request is ready for review.
- [ ] I think this pull request is ready to merge.

### CODE REVIEW

**Comment by MilesMcBain** [comment]
Some things to note:

* devtools::check() failed, but it seems due to something unrelated to my changes: 
> Error: package or namespace load failed for ‘drake’:
    object 'type_sum' not found whilst loading namespace 'drake
* The test I added might be over engineered. It tests a grid of 36 combinations of target name and code position and took 0.7s to run on my local test run with the longer tests taking around 3 - 4s. So it seems like a lot of time to test a small feature. That said I am confident it is pretty solid. :sunglasses:
  - * I deviated slightly for the architecture we discussed and pushed the context arg up to the main addin function. This meant I could test the whole addin.
* I have tried to use your preferred argument code style but I might not have nailed it in places. Sorry in advance. Happy to fix wherever necessary.

**Comment by MilesMcBain** [comment]
Ahh I see CI check has picked up missing `rstudioapi` in suggests. On that now.

**Comment by wlandau** [comment]
Thanks, Miles! I plan to review this tomorrow or Monday.

File: tests/testthat/test-rstudio-loadd.R [hunk 1]
```diff
@@ -0,0 +1,137 @@
+drake_context("rstudio")
+
+test_with_dir("loading targets at cursor works under a real conditions", {
+
```
Comment (wlandau) [inline tests/testthat/test-rstudio-loadd.R]: Could we add `skip_on_cran()` here? The CRAN gatekeepers have told me to cut down on testing runtime in previous submissions.

File: tests/testthat/test-rstudio-loadd.R [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  suppressMessages(
+    outcome <- test_loadd_start_end_middle(target_test_grid$test_text,
+                                           target_test_grid$target_start_col,
+                                           target_test_grid$target_length))
+
+  expect_identical(outcome, rep(TRUE, nrow(target_test_grid)))
+})
+
+test_with_dir("do nothing if the cursor is in the console", {
+  console_context <- structure(list(
```
Comment (wlandau) [inline tests/testthat/test-rstudio-loadd.R]: Same as above: I would like to add `skip_on_cran()` here.

File: R/rstudio-loadd.R [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  match_index <- which(
+    cursor_column >= match_starts &
+      cursor_column <= match_ends
+  )
+  substr(
+    context$contents[cursor_line],
+    start = match_starts[match_index],
+    stop = match_ends[match_index]
+  )
+}
```
Comment (wlandau) [inline R/rstudio-loadd.R]: Would you add a terminating line break to this file?

File: tests/testthat/test-rstudio-loadd.R [hunk 4]
```diff
@@ -0,0 +1,137 @@
+drake_context("rstudio")
```
Comment (wlandau) [inline tests/testthat/test-rstudio-loadd.R]: Could we change this to `drake_context("rstudio loadd")`?

**Comment by MilesMcBain** [comment]
Hi, I should be able to get to these over the weekend. I think I can sort out that error and the rest is no worries.

**Comment by wlandau** [comment]
Awesome, thanks Miles!

**Comment by MilesMcBain** [comment]
Ugh. I tried to rebase my branch onto master but now that has pulled in all your commits. I'll see if I can sort that out.

**Comment by MilesMcBain** [comment]
I don't think there's a way for me to fix this without starting a new PR - I checked at that will just contain my commits. Would you like me to do that?

File: tests/testthat/test-rstudio-loadd.R (hunk 1 above)
Comment (MilesMcBain) [reply tests/testthat/test-rstudio-loadd.R]: done

File: tests/testthat/test-rstudio-loadd.R (hunk 2 above)
Comment (MilesMcBain) [reply tests/testthat/test-rstudio-loadd.R]: done

File: R/rstudio-loadd.R (hunk 3 above)
Comment (MilesMcBain) [reply R/rstudio-loadd.R]: done

**Comment by wlandau** [comment]
Yeah, please feel free to start a new PR. The false positives under "Changed files" will make it hard for me to review anyway. Sorry about the mess.

**Pull request closed without merging**